app/controllers/em_website_controller.rb

```diff
@@ -12,10 +13,9 @@ def pricing
   def add_email
     if Person.email_create(params['person']['address'])
       flash[:notice] = "Thank you for signing up!"
-      redirect_back fallback_location: '/'
+
     else
-      flash[:alert] = "This email address has already been sign up."
-      redirect_back fallback_location: '/'
+      flash[:alert] = "The email entered has already been submitted."
```

Comment:
@SergeyProger what do you think about setting messages like these in the Person model, and then getting the message from there? For example:
```ruby
validates :email, uniqueness: true, message: "The email entered has already been submitted."
```
Then in the controller, something like this:
```ruby
person = Person.new(params) # Will need to change address to email in the form
person.valid? ? person.save : flash[:alert] = person.errors.first

redirect_back fallback_location: '/'
```
That way we can add more model level validations and the error messaging will work properly without having to rewrite this part.

Reply:
Thanks you. It works great!